tokio-stream/src/lib.rs

```diff
@@ -81,6 +81,10 @@ pub mod wrappers;
 
 mod stream_ext;
 pub use stream_ext::{collect::FromStream, StreamExt};
+pub use stream_ext::{
+    AllFuture, AnyFuture, Chain, Filter, FilterMap, FoldFuture, Fuse, Map, MapWhile, Merge, Next,
+    Peekable, Skip, SkipWhile, Take, TakeWhile, Then, TryNext,
+};
```

Comment:
Please put them in a sub-module so they don't clutter the front page of the crate documentation.

Reply:
I could define a `pub mod adapters` in lib.rs that `pub use`s all the structures like so
```rust
/// Some doc
pub mod adapters {
    pub use crate::stream_ext::{
        AllFuture, AnyFuture, Chain, Filter, FilterMap, FoldFuture, Fuse, Map, MapWhile, Merge,
        Next, Peekable, Skip, SkipWhile, Take, TakeWhile, Then, TryNext,
    };
}
```

This seemed very hacky to me, so I thought : should all the mods for those structures really be in the `stream_ext` folder/mod? The exposed structure of the crate is very different from its folder structure. How about having adapters.rs next to stream_ext.rs, and an adapters folder with all the mods' files? That would mimic the std::iter structure.
That way we could also put the "sources" (std::iter terminology) related mods (`empty`, `iter`, `once`, etc.) in their folder - but expose them the same as they are right now to not break anything.

I would have 2 questions with this though :
- is that considered a breaking change (the interface is kept the same)? And even if not, do you want to do this sort of big changes?
- what to do with the all-alone `collect` mod? Keeping it in `stream_ext`? Lift it next to `stream_ext` and `adapters`?

But perhaps I'm going too far, I have no idea :sweat_smile:

Reply:
Defining an inline module like you suggested in the beginning is fine. (But the other approaches would not be considered breaking.)